apisx/v1alpha1/xlistenerset_types.go

```diff
@@ -167,8 +167,6 @@ type ListenerEntry struct {
 	// +optional
 	//
 	// +kubebuilder:default=0
-	// +kubebuilder:validation:Minimum=0
-	// +kubebuilder:validation:Maximum=65535
```

Comment:
If we're going to remove these, then we need to move this validation to the aliased `PortNumber` type from `v1`, otherwise we're just _removing_ the validation altogether.

That does mean that we are tightening the validation on this field, technically, but since a value outside these bounds would not be programmable anyway, I think it's fine.

@robscott ?